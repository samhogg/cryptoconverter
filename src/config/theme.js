const theme = {
  colors: {
    primary: "#0062AE",
    error: "#be0000"
  },

  // System Font Stack
  // Adapted from https://github.com/mrmrs/css-system-fonts
  fontStack: `-apple-system, BlinkMacSystemFont,
    /* MacOS and iOS */ "avenir next", avenir, /* MacOS and iOS */ "Segoe UI",
    /* Windows */ "lucida grande", /* Older MacOS */ "helvetica neue", helvetica,
    /* Older MacOS */ "Fira Sans", /* Firefox OS */ roboto, noto,
    /* Google stuff */ "Droid Sans", /* Old Google stuff */ cantarell, oxygen,
    ubuntu, /* Linux stuff */ "franklin gothic medium", "century gothic",
    /* Windows stuff */ "Liberation Sans", /* Linux */ sans-serif; /* Everything else */`
};

export default theme;
