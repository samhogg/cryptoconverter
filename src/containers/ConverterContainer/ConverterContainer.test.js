import React from "react";
import renderer from "react-test-renderer";
import { ConverterContainer } from ".";
import { ThemeProvider } from "styled-components";
import theme from "../../config/theme";

it("Renders without errors", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ConverterContainer
          balances={{ usd: 100, btc: 100 }}
          btcValue={0}
          loading={true}
          executeTrade={() => {}}
          getQuote={() => {}}
          updateQuoteValue={() => {}}
          usdValue={0}
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("Renders with an error", () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <ConverterContainer
          balances={{ usd: 100, btc: 100 }}
          btcValue={0}
          error="Fetch Error"
          executeTrade={() => {}}
          getQuote={() => {}}
          updateQuoteValue={() => {}}
          usdValue={0}
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
