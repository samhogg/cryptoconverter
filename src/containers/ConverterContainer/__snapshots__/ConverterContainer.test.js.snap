// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with an error 1`] = `
<div
  className="styledComponents__Container-kJQGqy gwKzdI"
>
  <div
    className="styledComponents__ErrorWrapper-bxVaWE iTsRMw"
  >
    Fetch Error
    <button
      className="styledComponents__RetryButton-eBsaHs iRHKIs"
      onClick={[Function]}
    >
      Retry
    </button>
  </div>
  <section
    className="styledComponents__BalanceContainer-cIYnGy iIjgYz"
  >
    <h2
      className="styledComponents__BalanceListTitle-bPaUMl hDojKG"
    >
      Account Balance
    </h2>
    <ul
      className="styledComponents__BalanceList-bqFABO eHXqVh"
    >
      <li
        className="styledComponents__BalanceItem-fDtCqf gzMZya"
      >
        <div
          className="styledComponents__BalanceTicker-dDjLel bpTZSY"
        >
          usd
        </div>
        <div
          className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
        >
          <span
            className="styledComponents__BalancePrefix-fCshfk eIVmaR"
          >
            $
          </span>
          100.00
        </div>
      </li>
      <li
        className="styledComponents__BalanceItem-fDtCqf gzMZya"
      >
        <div
          className="styledComponents__BalanceTicker-dDjLel bpTZSY"
        >
          btc
        </div>
        <div
          className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
        >
          <span
            className="styledComponents__BalancePrefix-fCshfk eIVmaR"
          >
            ₿
          </span>
          100.000000
        </div>
      </li>
    </ul>
  </section>
  <form
    className="styledComponents__TradeContainer-kAVRwM ZpDir"
  >
    <h2
      className="styledComponents__TradeTitle-chyUqE iUFiLO"
    >
      Trade
    </h2>
    <div
      className="LoadingIndicator__LoadingContainer-eMCbod fjLIAU"
    >
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
    </div>
    <input
      className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
      disabled={false}
      onChange={[Function]}
      placeholder="USD"
      readOnly={false}
      value=""
    />
    <div
      className="styledComponents__TradeSeparator-faTACV gxVESP"
    >
      To
    </div>
    <input
      className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
      disabled={true}
      onChange={undefined}
      placeholder="BTC"
      readOnly={true}
      value=""
    />
    <button
      className="styledComponents__TradeButton-ilQyaa ezRzaG"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Execute Trade
    </button>
  </form>
</div>
`;

exports[`Renders without errors 1`] = `
<div
  className="styledComponents__Container-kJQGqy cbPYLL"
>
  <section
    className="styledComponents__BalanceContainer-cIYnGy iIjgYz"
  >
    <h2
      className="styledComponents__BalanceListTitle-bPaUMl hDojKG"
    >
      Account Balance
    </h2>
    <ul
      className="styledComponents__BalanceList-bqFABO eHXqVh"
    >
      <li
        className="styledComponents__BalanceItem-fDtCqf gzMZya"
      >
        <div
          className="styledComponents__BalanceTicker-dDjLel bpTZSY"
        >
          usd
        </div>
        <div
          className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
        >
          <span
            className="styledComponents__BalancePrefix-fCshfk eIVmaR"
          >
            $
          </span>
          100.00
        </div>
      </li>
      <li
        className="styledComponents__BalanceItem-fDtCqf gzMZya"
      >
        <div
          className="styledComponents__BalanceTicker-dDjLel bpTZSY"
        >
          btc
        </div>
        <div
          className="styledComponents__BalanceQuantity-kFtsKU jMHtFP"
        >
          <span
            className="styledComponents__BalancePrefix-fCshfk eIVmaR"
          >
            ₿
          </span>
          100.000000
        </div>
      </li>
    </ul>
  </section>
  <form
    className="styledComponents__TradeContainer-kAVRwM ZpDir"
  >
    <h2
      className="styledComponents__TradeTitle-chyUqE iUFiLO"
    >
      Trade
    </h2>
    <div
      className="LoadingIndicator__LoadingContainer-eMCbod bZALBS"
    >
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
      <div
        className="LoadingIndicator__LoadingDot-kbVGSy eVeAmw"
      />
    </div>
    <input
      className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
      disabled={false}
      onChange={[Function]}
      placeholder="USD"
      readOnly={false}
      value=""
    />
    <div
      className="styledComponents__TradeSeparator-faTACV gxVESP"
    >
      To
    </div>
    <input
      className="TradeCurrencyPair__TradeInput-bFnQSV eEMErO"
      disabled={true}
      onChange={undefined}
      placeholder="BTC"
      readOnly={true}
      value=""
    />
    <button
      className="styledComponents__TradeButton-ilQyaa ezRzaG"
      disabled={true}
      onClick={[Function]}
      type="submit"
    >
      Execute Trade
    </button>
  </form>
</div>
`;
